wrapper/src/test/java/software/amazon/jdbc/util/WrapperUtilsTest.java

```diff
@@ -144,6 +151,30 @@ void testExecutesWithPluginsWithExceptionIsSequential() {
     }
   }
 
+  @Test
+  void testCancelStatementIsNotBlockedExecute() {
+    List<CompletableFuture<Integer>> futures = new ArrayList<>();
+
+    futures.add(CompletableFuture.supplyAsync(this::callExecuteWithPlugins));
+    futures.add(CompletableFuture.supplyAsync(this::callCancelExecuteWithPlugins));
+
+    for (CompletableFuture<Integer> future : futures) {
+      future.join();
+    }
+  }
```

Comment:
If you don't care about the results just do
```suggestion
    CompletableFuture.allOf(
        CompletableFuture.supplyAsync(this::callExecuteWithPluginsWithException),
        CompletableFuture.supplyAsync(this::callCancelExecuteWithPluginsWithException)
    ).join();
```